src/textual/_animator.py

```diff
@@ -374,6 +426,65 @@ def _animate(
         self._idle_event.clear()
         self._complete_event.clear()
 
+    async def _stop_scheduled_animation(
+        self, key: AnimationKey, complete: bool
+    ) -> None:
+        """Stop a scheduled animation.
+
+        Args:
+            key: The key for the animation to stop.
+            complete: Should the animation be moved to its completed state?
+        """
+        # First off, pull the timer out of the schedule and stop it; it
+        # won't be needed.
+        schedule = self._scheduled[key]
```

Comment:
Danger Will Robinson. As this is async, a ton of code could have run since the caller awaited this method. Enough time for `key` to be removed in another task.

Reply:
A bit more detail about your thinking here would be helpful. For example, are you suggesting that it would be a good idea to handle the key not being in the dictionary any more, because the animation may have elsewhere run to completion in the moment between the call to stop and this method getting hit?

Reply:
Exactly, you can't assume that key will still be in that dict.

Reply:
Cool. Handled.